metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -669,7 +669,11 @@ class MetalsLanguageServer(
       event: DirectoryChangeEvent
   ): CompletableFuture[Unit] = {
     val path = AbsolutePath(event.path())
-    if (!savedFiles.isRecentlyActive(path) && path.isScalaOrJava) {
+    val isScalaOrJava = path.isScalaOrJava
+    if (isScalaOrJava && event.eventType() == EventType.DELETE) {
+      diagnostics.didDelete(path)
```

Comment:
We already have tests to see that diagnostics are cleared for deleted files after a `didSave` notification. I manually verified that this works as soon as the file is deleted without a `didSave` notification. However, I don't think we can write a non-flaky test case to stress this situation since file watchers are quite unreliable between operating systems.